### PR TITLE
security: prevent last-admin lockout on user demotion and deletion

### LINKED ISSUE #209: security: prevent last-admin lockout (self-demotion / deletion of the final admin)
## Problem

Found during the #187 RBAC review (PR #208). The role-enforcement work makes user administration admin-only, but nothing prevents the **last remaining admin** from removing their own access:

- An admin can demote themselves: `PUT /api/v1/users/:id` with `{"role": "contributor"}` on their own ID is allowed (admins may change any role, including their own).
- An admin can delete any admin account, including their own or the only other one: `DELETE /api/v1/users/:id` has no last-admin guard.

If the only admin is demoted or deleted, no account can manage users, themes, settings, or post types. Recovery requires direct SQL or the debug-mode default-user seeder (`devModeUtil/create_default_user.go`, dev builds only) - nothing recoverable from the API itself.

## What Needs to Happen

In `api/users_handlers_write.go`:

1. **`updateUser`**: when the request changes `role` away from admin AND the target is currently an admin, count remaining admins; reject with 409 (or 403 + clear message) if the target is the last one.
2. **`deleteUser`**: same guard - refuse to delete the last admin account.
3. Needs a small query, e.g. `CountUsersByRole(role)` in `db/query/users.sql` + `sqlc generate` (no schema change).

Edge cases to cover in tests:
- Last admin demoting themselves -> rejected
- Last admin deleting themselves -> rejected
- Demoting/deleting an admin while another admin exists -> allowed
- Two concurrent "demote the other admin" requests (race) - acceptable to leave as known limitation if documented, or guard transactionally

## Acceptance Criteria

- [ ] API refuses to demote or delete the final admin-role user
- [ ] Clear error message distinguishes this from a generic forbidden
- [ ] Handler tests cover the last-admin and not-last-admin paths

## Related

- #187 / PR #208 (RBAC enforcement - introduced admin-gated user management)
- `api/users_handlers_write.go` - updateUser / deleteUser
- `db/query/users.sql` - new count query

### PR DESCRIPTION
**Pull request by luan-k**
Closes #209. Stacked on #208 (base: `187-rbac-enforcement`) - only the last-admin commit is in this diff; retarget to `main` after #208 merges.  ## What this does  #208 made user, theme, settings, and post-type management admin-only - but nothing stopped the **final admin** from removing their own access. An admin could demote themselves (`PUT /users/:id` with a non-admin `role`) or delete the only admin account (`DELETE /users/:id`), leaving the site with no account able to administer it. Recovery required direct SQL or the debug-mode default-user seeder.  Both handlers in `api/users_handlers_write.go` now guard the operation:  - **`updateUser`**: when the request moves an admin target OFF the admin role, count remaining admins; if the target is the last one -> `409 Conflict` (`"cannot demote the last admin"`). - **`deleteUser`**: same guard when the delete target is an admin -> `409 Conflict` (`"cannot delete the last admin"`). - Demoting or deleting an admin while another admin exists is unaffected. Role comparison stays case-insensitive end to end (new `CountUsersByRole` sqlc query compares `lower(role)`).  `409` rather than `403` because the caller IS authorized - the operation conflicts with system state, and the explicit message tells the admin exactly why it was refused.  **Known limitation (documented in the code):** the admin count runs outside the mutation transaction, so any pair of concurrent admin-removing operations (demote+demote, demote+delete) can each observe count=2 and end at zero admins - not just concurrent demotes. Accepted for now; the structural fix is a count with FOR UPDATE semantics inside UpdateUserTx/DeleteUserTx if it ever matters in practice.  ## Changes  - `db/query/users.sql` + regenerated `db/sqlc/` and `db/mock/store.go`: new `CountUsersByRole` query (sqlc v1.29.0 / mockgen v1.6.0, matching committed generator versions - zero churn outside the new method) - `api/users_handlers_write.go`: last-admin guards in `updateUser` / `deleteUser`, doc comments updated with the 409 responses  ## Testing  - `go test ./...` green, including `db/sqlc` against real Postgres - New mock cases: demote last admin -> 409; delete last admin -> 409; demote/delete an admin while a second admin exists -> 200; existing update/delete/403 cases unchanged - New real-DB test for `CountUsersByRole` incl. case-insensitivity - Live check against the dev stack: demoting `testadmin` while other admins exist succeeds and restores cleanly (guard path executes the real query)  ## Manual testing (for the reviewer)  The dev DB currently has three admins (`admin`, `LuanK`, `testadmin` - password `123456` for `admin`/`testadmin`). To see the 409 you need to get down to one admin:  ```bash TOK=$(curl -s -X POST http://localhost:8080/api/v1/auth/login \   -H "Content-Type: application/json" \   -d '{"username":"admin","password":"123456"}' | jq -r .access_token)  # Demote the other admins first (allowed while >1 admin remains), e.g. testadmin (id 1812): curl -s -X PUT http://localhost:8080/api/v1/users/1812 \   -H "Authorization: Bearer $TOK" -H "Content-Type: application/json" \   -d '{"role":"editor"}'  # ...repeat for any other admin account, then try to demote yourself (admin is id 1) -> 409: curl -s -w "\n%{http_code}\n" -X PUT http://localhost:8080/api/v1/users/1 \   -H "Authorization: Bearer $TOK" -H "Content-Type: application/json" \   -d '{"role":"contributor"}'  # And deleting yourself as the last admin -> 409: curl -s -w "\n%{http_code}\n" -X DELETE http://localhost:8080/api/v1/users/1 \   -H "Authorization: Bearer $TOK" -H "Content-Type: application/json" -d '{}'  # Restore the demoted admins afterwards: curl -s -X PUT http://localhost:8080/api/v1/users/1812 \   -H "Authorization: Bearer $TOK" -H "Content-Type: application/json" \   -d '{"role":"admin"}' ```  🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: api/users_handlers_write.go [hunk 1]
```diff
@@ -165,6 +165,23 @@ func (server *Server) updateUser(c *gin.Context) {
 		return
 	}
 
+	// Last-admin guard: demoting the only remaining admin would leave the
+	// site with no account able to manage users, themes, or settings.
+	demotingAdmin := req.Role != "" &&
+		strings.EqualFold(existingUser.Role, RoleAdmin) &&
+		!strings.EqualFold(req.Role, RoleAdmin)
```
[the remark this answers is not in the document]
Comment (luan-k) [reply api/users_handlers_write.go:177]: Fixed in d7fdd81: both guard comments now list post types alongside users, themes, and settings.

File: api/users_handlers_write.go [hunk 2]
```diff
@@ -264,6 +282,20 @@ func (server *Server) deleteUser(c *gin.Context) {
 		return
 	}
 
+	// Last-admin guard: deleting the only remaining admin would leave the
+	// site with no account able to manage users, themes, or settings.
+	if strings.EqualFold(targetUser.Role, RoleAdmin) {
```
[the remark this answers is not in the document]
Comment (luan-k) [reply api/users_handlers_write.go:294]: Fixed in d7fdd81 (same commit as the updateUser comment).
